source/Calamari.AzureAppService/Behaviors/Factories/AzureAppServiceDeployBehaviourFactory.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Threading.Tasks;
+using Calamari.Common.Commands;
+using Calamari.Common.FeatureToggles;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Pipeline;
+
+namespace Calamari.AzureAppService.Behaviors
+{
+    public class AzureAppServiceDeployBehaviourFactory : IDeployBehaviour
+    {
+        readonly ILog log;
+
+        public AzureAppServiceDeployBehaviourFactory(ILog log)
+        {
+            this.log = log;
+        }
+        
+        public bool IsEnabled(RunningDeployment context) 
+            => true;
+
+        public Task Execute(RunningDeployment context)
+        {
+            return FeatureToggle.UseModernAzureAppServiceSdkFeatureToggle.IsEnabled(context.Variables) 
+                ? new AzureAppServiceBehaviour(log).Execute(context)
+                : new LegacyAzureAppServiceBehaviour(log).Execute(context);
+        }
```

Comment:
This is an example of a factory behaviour that uses the feature toggle to swap between the new behaviour and the legacy behaviour